src/l1-resources/access-list-api-key/index.ts

```diff
@@ -0,0 +1,117 @@
+// Generated by cdk-import
+import * as cdk from "aws-cdk-lib";
+import * as constructs from "constructs";
+
+/**
+ * Creates the access list entries for the specified organization API key.
+ *
+ * @schema CfnAccessListApiKeyProps
+ */
+export interface CfnAccessListApiKeyProps {
+  /**
+   * Unique 24-hexadecimal digit string that identifies the organization that contains your projects
+   *
+   * @schema CfnAccessListApiKeyProps#OrgId
+   */
+  readonly orgId: string;
+
+  /**
+   * Unique 24-hexadecimal digit string that identifies this organization API key for which you want to return access list entries.
+   *
+   * @schema CfnAccessListApiKeyProps#ApiUserId
+   */
+  readonly apiUserId: string;
+
+  /**
+   * Network address that issued the most recent request to the API.
+   *
+   * @schema CfnAccessListApiKeyProps#Profile
+   */
+  readonly profile?: string;
+
+  /**
+   * Range of network addresses that you want to add to the access list for the API key.
+   *
+   * @schema CfnAccessListApiKeyProps#CidrBlock
+   */
+  readonly cidrBlock?: string;
+
+  /**
+   * Network address that you want to add to the access list for the API key.
+   *
+   * @schema CfnAccessListApiKeyProps#IpAddress
+   */
+  readonly ipAddress?: string;
+
+  /**
+   * Number of documents returned in this response.
+   *
+   * @schema CfnAccessListApiKeyProps#TotalCount
+   */
+  readonly totalCount?: number;
+}
+
+/**
+ * Converts an object of type 'CfnAccessListApiKeyProps' to JSON representation.
+ */
+/* eslint-disable max-len, quote-props */
+export function toJson_CfnAccessListApiKeyProps(
```

Comment:
Now that I am reviewing these PRs more frequently I see these type of methods in a lot of them. I really wonder if we could do something better here, writing one only of these methods that could be unit tested and passing the result as input or using some form of type abstraction that Typescript should have available.

Only in this PR we also have `toJson_ProjectAssignment` and `toJson_CfnApiKeyProps` --they are all conceptually the same, only thing it changes is the name of the fields in the result variable.

Code duplication is not necessarily bad, in this case though I see a potential huge savings for code reviews and less human errors.

Reply:
i wouldnt suggest to change this functions, this are autimaticly generated by the cdk-import tool, you can see it in the script that is in scripts/cdk.sh